src/auth/Crypto.h

```diff
@@ -140,6 +140,8 @@ class CryptoHandler {
   virtual int validate_secret(const bufferptr& secret) = 0;
   virtual CryptoKeyHandler *get_key_handler(const bufferptr& secret,
 					    string& error) = 0;
+  virtual CryptoKeyHandler *get_key_handler(const bufferptr& secret, const bufferptr& iv,
+              string& error) = 0;
```

Comment:
@cbodley I like what you propose, but the problem is I cannot give default value to iv unless I give it also for next parameter (`string& error`). 
Maybe it would be best if I make one `get_key_handler` regular, implemented by invoke of the other one with default iv supplied? 
Like: 
`CryptoKeyHandler *get_key_handler(const bufferptr& secret, string& error) {
return get_key_handler(secret, ceph_iv, error); 
}`
 ?